### ISSUE TITLE
[photonics-service] Test coverage is low (Trac #1272)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1266]
...surprisingly low.  Maybe we need to go through this project and figure out what we still need now that we're not simulating with photonics like we used to.

Migrated from https://code.icecube.wisc.edu/ticket/1272
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:11:57", 
    "description": "...surprisingly low.  Maybe we need to go through this project and figure out what we still need now that we're not simulating with photonics like we used to.", 
    "reporter": "olivas", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067117911749", 
    "component": "combo simulation", 
    "summary": "[photonics-service] Test coverage is low", 
    "priority": "major", 
    "keywords": "", 
    "time": "2015-08-21T16:52:15", 
    "milestone": "", 
    "owner": "andrii.terliuk", 
    "type": "enhancement"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20150908T19:48:18: olivas changed status from "new" to "accepted"

**Comment by icecube-trac** [comment]
Trac update at 20151007T16:18:10: 
* olivas changed owner from "olivas" to "andrii.terliuk"
* olivas commented:

> I was considering pushing this to the next milestone, but if it's *really* low we should be able to make some progress, right?  Let's get one or two more tests in and push the milestone again.  We don't need 100% coverage immediately.


* olivas changed status from "accepted" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20160215T12:31:24: kjmeagher commented:

> Most of the stuff in I3Photonics*Reader.cxx is for generating photonics based simulation. If it is not needed anymore it can be obliterated.

**Comment by icecube-trac** [comment]
Trac update at 20160215T15:31:17: olivas commented:

> Is it not used in any recos?
> 
> Also, let's make sure we're currently simulating tables just fine with clsim before we set anything on fire.

**Comment by icecube-trac** [comment]
Trac update at 20160215T15:36:51: kjmeagher commented:

> It is used in truncated energy, but that only uses a small fraction of the code

**Comment by icecube-trac** [comment]
Trac update at 20160216T13:45:34: kjmeagher commented:

> r141855/IceCube should provide test coverage for everything that is actually still needed. What is left was only needed for hitmaker

**Comment by icecube-trac** [comment]
Trac update at 20160216T16:53:12: olivas commented:

> Can we remove that too?  I'd be surprised if hit-maker built out of the box with the current trunk. There's no longer support in production for generating hit-maker samples.

**Comment by icecube-trac** [comment]
Trac update at 20160217T16:19:41: kjmeagher commented:

> I wrote a unit test for photonics which simulates a nearly horizontal muon so that it only takes one zenith bin and 4 z bins to use, but it still needs 260 MB of tables to work

**Comment by icecube-trac** [comment]
Trac update at 20160219T17:13:58: 
* kjmeagher changed status from "assigned" to "closed"
* kjmeagher changed resolution from "" to "fixed"
* kjmeagher commented:

> In changeset 142123:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="142123"
> add unit tests for I3PhotonicsHitMaker using both splines and tables
> this get coverage up to around 50% which closes #1266
> The remaining code is a combination of  total crap (a bad implementation of endian swap, a fallback reader incase mmap fails, and support for spherical and cartesian tables) as well as legitimate functions which don't seem to be called by any current IceTray Module
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:11:57: 
* olivas commented:

> Milestone IceRec V5 deleted


* olivas changed milestone from "IceRec V5" to ""
